### ISSUE TITLE
"Instruction does not dominate all uses!" with `opt "-passes=loop-mssa(licm<allowspeculation>,loop-deletion,simple-loop-unswitch<nontrivial>),loop(indvars)"`

### ISSUE THREAD

**Issue by mikaelholmen** [llvm/llvm-project#58871]
LLVM commit: e661185fb3e5
Reproduce with:
```shell
opt "-passes=loop-mssa(licm<allowspeculation>,loop-deletion,simple-loop-unswitch<nontrivial>),loop(indvars)" bbi-75601_2.ll -o /dev/null
```
Result:
```console
Instruction does not dominate all uses!
  %f.b.promoted.lcssa2 = phi i16 [ %f.b.promoted, %for.body.preheader ]
  %2 = sub i16 %.us-phi, %f.b.promoted.lcssa2
LLVM ERROR: Broken module found, compilation aborted!
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: ../../main-github/llvm/build-all/bin/opt -passes=loop-mssa(licm<allowspeculation>,loop-deletion,simple-loop-unswitch<nontrivial>),loop(indvars) bbi-75601_2.ll -o /dev/null
 #0 0x0000000002e63883 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (../../main-github/llvm/build-all/bin/opt+0x2e63883)
 #1 0x0000000002e6159e llvm::sys::RunSignalHandlers() (../../main-github/llvm/build-all/bin/opt+0x2e6159e)
 #2 0x0000000002e63c06 SignalHandler(int) Signals.cpp:0:0
 #3 0x00007f9abd677630 __restore_rt sigaction.c:0:0
 #4 0x00007f9abadbe387 raise (/lib64/libc.so.6+0x36387)
 #5 0x00007f9abadbfa78 abort (/lib64/libc.so.6+0x37a78)
 #6 0x0000000002dd355b llvm::report_fatal_error(llvm::Twine const&, bool) (../../main-github/llvm/build-all/bin/opt+0x2dd355b)
 #7 0x0000000002dd3378 (../../main-github/llvm/build-all/bin/opt+0x2dd3378)
 #8 0x00000000026a4a14 (../../main-github/llvm/build-all/bin/opt+0x26a4a14)
 #9 0x000000000072a3ad llvm::detail::PassModel<llvm::Module, llvm::VerifierPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) crtstuff.c:0:0
#10 0x000000000266f79c llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (../../main-github/llvm/build-all/bin/opt+0x266f79c)
#11 0x0000000000722c49 llvm::runPassPipeline(llvm::StringRef, llvm::Module&, llvm::TargetMachine*, llvm::TargetLibraryInfoImpl*, llvm::ToolOutputFile*, llvm::ToolOutputFile*, llvm::ToolOutputFile*, llvm::StringRef, llvm::ArrayRef<llvm::StringRef>, llvm::ArrayRef<llvm::PassPlugin>, llvm::opt_tool::OutputKind, llvm::opt_tool::VerifierKind, bool, bool, bool, bool, bool, bool) (../../main-github/llvm/build-all/bin/opt+0x722c49)
#12 0x0000000000732301 main (../../main-github/llvm/build-all/bin/opt+0x732301)
#13 0x00007f9abadaa555 __libc_start_main (/lib64/libc.so.6+0x22555)
#14 0x000000000071b8b0 _start (../../main-github/llvm/build-all/bin/opt+0x71b8b0)
Abort (core dumped)
```
[bbi-75601_2.ll.gz](https://github.com/llvm/llvm-project/files/9959113/bbi-75601_2.ll.gz)

**Comment by mikaelholmen** [comment]
This starts happening with 20d798bd47ec
```Recommit "[SCEV] Look through single value PHIs." (take 3)```

**Comment by fhahn** [comment]
Looks like this is the same underlying issue as #58868 and should be fixed by 7854a1abfd9a52dec0fe5d668e26de19614faefb
